example/src/screens/Preview.tsx

```diff
@@ -0,0 +1,232 @@
+} from 'react';
+import { View, StyleSheet, Alert } from 'react-native';
+import { useVideoroomState } from 'src/VideoroomContext';
+
+type Props = NativeStackScreenProps<RootStack, 'Preview'>;
+
+export const Preview = ({ navigation, route }: Props) => {
+  const { roomName, setRoomName, username, setUsername } = useVideoroomState();
+  const [isCameraOn, setIsCameraOn] = useState(true);
+  const [isMicrophoneOn, setIsMicrophoneOn] = useState(true);
+  const isSimulcastOn = true;
+
+  const { prevScreen } = route.params;
+  const headerTitle =
+    prevScreen === 'CreateRoom' ? 'New meeting' : 'Join meeting';
+  const titleLabel =
+    prevScreen === 'CreateRoom'
+      ? 'Create a new room to start the meeting'
+      : 'Join an existing meeting';
+
+  const { connect: mbConnect, joinRoom, error } = Membrane.useMembraneServer();
+
+  const params = {
+    token: 'NOW_YOU_CAN_SEND_PARAMS',
+  };
+
+  useEffect(() => {
+    if (error) {
+      console.log(error);
+      Alert.alert('Error when connecting to server:', error);
+    }
+  }, [error]);
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      title: headerTitle,
+    });
+  }, [navigation]);
+
+  const getShortUsername = () => {
+    return username
+      .split(' ')
+      .map((i) => i.charAt(0))
+      .join('')
+      .toUpperCase();
+  };
+
+  const connect = useCallback(async () => {
+    const validRoomName = roomName.trimEnd();
+    const validUserName = username.trimEnd();
+
+    setRoomName(validRoomName);
+    setUsername(validUserName);
+
+    try {
+      await mbConnect(SERVER_URL, validRoomName, {
+        userMetadata: { displayName: validUserName },
+        connectionParams: params,
+        socketChannelParams: {
+          isSimulcastOn,
+        },
+        simulcastConfig: {
+          enabled: isSimulcastOn,
+          activeEncodings: ['l', 'm', 'h'],
+        },
+        quality: Membrane.VideoQuality.HD_169,
+        maxBandwidth: { l: 150, m: 500, h: 1500 },
+        videoTrackMetadata: { active: true, type: 'camera' },
+        audioTrackMetadata: { active: true, type: 'audio' },
+        isSpeakerphoneOn: false,
+        videoTrackEnabled: isCameraOn,
+        audioTrackEnabled: isMicrophoneOn,
+      });
+      await joinRoom();
+      navigation.navigate('Room');
+    } catch (err) {
+      console.warn(err);
+    }
+  }, [
+    mbConnect,
+    joinRoom,
+    roomName,
+    isSimulcastOn,
+    username,
+    SERVER_URL,
+    isCameraOn,
+    isMicrophoneOn,
+  ]);
+
+  return (
+    <BackgroundWrapper hasHeader>
+      <View style={styles.content}>
+        <View style={styles.header}>
+          <Typo variant="h4">Videoconferencing for everyone</Typo>
+        </View>
+        <View style={styles.titleLabel}>
+          <Typo variant="chat-regular">{titleLabel}</Typo>
+        </View>
+
+        <View style={styles.cameraPreview}>
+          {isCameraOn ? (
+            <Membrane.VideoPreviewView style={styles.membraneVideoPreview} />
```

Comment:
Also the preview should be probably mirrored for the front camera, because people are used to looking at their faces in the mirror.